multiqc/modules/seqyclean/seqyclean.py

```diff
@@ -0,0 +1,88 @@
+# !/usr/bin/env python
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+import re
+
+from multiqc import config
+from multiqc.plots import linegraph, bargraph, table
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+	def __init__(self):
+		# Initialise the parent object
+		super(MultiqcModule, self).__init__(name='SeqyClean',
+		anchor='seqyclean',
+		href='https://github.com/ibest/seqyclean',
+		info='SeqyClean is a pre-processing tool for NGS data that filters adapters, vectors, and contaminants while quality trimming.')
+
+		# Parse logs
+		self.seqyclean_data = dict()
+		for f in self.find_log_files('seqyclean'):
+			rows = f['f'].splitlines()
+			headers = rows[0].split("\t")
+			cols = rows[1].split("\t")
+			f['s_name'] = re.sub('_SummaryStatistics','',f['s_name'])
+
+			self.seqyclean_data[f['s_name']] = dict()
+			for header, cols in zip(headers, cols):
+				self.seqyclean_data[f['s_name']].update({ header : cols })
+
+		if len(self.seqyclean_data) == 0:
+			raise UserWarning
+
+		self.seqyclean_data = self.ignore_samples(self.seqyclean_data)
+
+		log.info("Found {} logs".format(len(self.seqyclean_data)))
+
+		# Adding the bar plot
+		self.add_section(plot = self.seqyclean_bargraph())
+
+		# Write the results to a file
+		self.write_data_file(self.seqyclean_data, 'seqyclean')
+
+		# Adding to the general statistics table
+		self.seqyclean_general_stats_table()
+
+	def seqyclean_bargraph(self):
+		config = {
+			'id': 'seqyclean-1',
+			'title': 'SeqyClean: Reads Analysis',
+			'ylab': 'Number of Reads'
+			}
+		keys =['PE1ReadsAn',
```

Comment:
This is going to take me a bit to resolve. Turns out that single-read has a different table than paired end. I need to find some Roche 454 data to use this on as well, because I bet the table is different there too.

Reply:
Yeah, I was worried that might be the case. I've hit similar problems with other tools. Don't worry too much about the 454 data as that's super old now, but single end will probably still be a fairly common use case for people.

Reply:
I grabbed some 454 data off of bioproject.